src/linker/Linker.Dataflow/MethodProxy.cs

```diff
@@ -32,7 +32,18 @@ namespace ILLink.Shared.TypeSystemProxy
 
 		internal partial bool IsStatic () => Method.IsStatic;
 
-		internal partial bool ReturnsVoid () => Method.ReturnType.MetadataType == MetadataType.Void;
+		/// <summary>
+		/// Uses the logic from <see cref="MethodBodyScanner.GetReturnTypeWithoutModifiers(TypeReference)"/> to get the
+		/// return type and determine if it is void
```

Comment:
Why not call that method directly?
We could either make it internal on the `MethodBodyScanner` or we could move it and make it a` TypeReference` extension.